### PR TITLE
implement delegate methods from https://github.com/brave-intl/bat-nat…

### PR DESCRIPTION
**Pull request by bridiver**
…ive-ledger/pull/11

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: browser/payments/payments_service_impl.cc [hunk 1]
```diff
@@ -3,23 +3,267 @@
...
+}
+
+void PaymentsServiceImpl::OnReconcileComplete(ledger::Result result,
+                                              const std::string& viewing_id) {
+  LOG(ERROR) << "reconcile complete " << viewing_id;
+}
+
+void PaymentsServiceImpl::LoadLedgerState(
+    ledger::LedgerCallbackHandler* handler) {
+  base::PostTaskAndReplyWithResult(file_task_runner_.get(), FROM_HERE,
```
Comment (bridiver) [inline browser/payments/payments_service_impl.cc]: static method and weakptr ensure that pending callbacks won't run on objects that have been destroyed

File: browser/payments/payments_service_impl.cc [hunk 2]
```diff
@@ -3,23 +3,267 @@
...
+void PaymentsServiceImpl::CreateWallet() {
+  ledger_->CreateWallet();
+}
+
+std::string PaymentsServiceImpl::GenerateGUID() const {
+  return base::GenerateGUID();
+}
+
+void PaymentsServiceImpl::Shutdown() {
+  fetchers_.clear();
```
Comment (bridiver) [inline browser/payments/payments_service_impl.cc]: fetchers is cleared before ledger is destroyed to ensure that no pending network callbacks run on destroyed objects

**Comment by bridiver** [comment]
just bumped the native-ledger ref since @SergeyZhukovsky's approval so merging

**Comment by bbondy** [comment]
lgtm

**Pull request merged**